crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -342,6 +345,11 @@ pub unsafe trait QueryData: WorldQuery {
         entity: Entity,
         table_row: TableRow,
     ) -> Option<Self::Item<'w, 's>>;
+
+    /// Returns an iterator over the access needed by [`QueryData::fetch`]. Access conflicts are usually
+    /// checked in [`WorldQuery::init_fetch`], but in certain cases this method can be useful to implement
```

Comment:
The access checks are in `update_component_access`, not `init_fetch`.  

```suggestion
    /// checked in [`WorldQuery::update_component_access`], but in certain cases this method can be useful to implement
```

(Hmm, I wonder whether it would be possible, as a follow-up, to implement `update_component_access` in terms of `iter_access`.  That might avoid needing to declare the access in two different ways.)